test/hack/cleanup/main.go

```diff
@@ -93,3 +107,30 @@ func main() {
 		}
 	}
 }
+
+func cleanUpCluster(ctx context.Context, name string, cfg aws.Config, logger *zap.SugaredLogger) bool {
+	soakExpirationTime := time.Now().Add(-soakExpirationTTL)
+	eksClient := eks.NewFromConfig(cfg)
+
+	if strings.HasPrefix(name, "soak-periodic-") {
+		cluster, err := eksClient.DescribeCluster(ctx, &eks.DescribeClusterInput{Name: aws.String(name)})
+		if err != nil {
+			logger.Errorf("%v", err)
+			return false
+		}
+
+		if !lo.FromPtr(cluster.Cluster.CreatedAt).Before(soakExpirationTime) {
+			logger.Infof("soak testing cluster (%s) does not need to be cleaned up until %s", name, soakExpirationTime)
+			return true
+		}
+	}
+
+	deleteCluster := exec.Command("eksctl", "delete", "cluster", "--name", name, "--timeout", "60m", "--wait")
```

Comment:
I don't think that we should be `exec`-ing out to `eksctl`. We should work to make this soak testing portable, so reduce the dependency sprawl that's required on the system that's running it.